feat(text): DLT-3487 expose full typography size scale (#1334)
dialpad/dialtone: apps/dialtone-documentation/docs/components/text.md

---
title: Text
description: Consistent typography styling through text variants and raw font-size tokens.

> status: ready

Reviewer: ```suggestion
status: new
````

bring this back to `new`

- Use `variant` for complete text compositions and pair `size` with `variant` only when a raw font-size override is needed.
- Most properties are optional, as they layer in on top of each other.

<dialtone-usage>
<template #do>

- Replace multiple `d-` typography classes with a single `dt-text` instance.
- Pick the smallest `variant` that conveys the desired hierarchy.
- Pair `size` with `variant` when the composition is right but the font size needs a token-backed override.
- Use `tone` for semantic color tokens instead of standalone `d-fc-*` classes.

</template>
<template #dont>

- Mix `DtText` with conflicting typography utilities (e.g., `d-fs-*`).
- Render headings with non-heading tags (e.g., avoid `as="div"` for top-level titles).
- Depend on the `text` prop when the content requires inline formatting; slot it instead.

</template>
</dialtone-usage>

## Variants

### Variant

Use `variant` to apply a complete text composition, including font family, font size, font weight, and line height. Default will inherit styles from the parent.

```vue demo
<!-- @wrapper -->
<dt-stack
  gap="400"
  :direction="{ default: 'column', md: 'row' }"
  align="baseline"
>

>   <dt-text variant="headline-md" as="span">Headline</dt-text>

Reviewer: ```suggestion
  <dt-text variant="headline-md" as="h3">Headline</dt-text>
```
Nit: change `as` to an `<h*>` element

  <dt-text variant="body-md">Body</dt-text>
  <dt-text variant="label-md">Label</dt-text>
  <dt-text variant="code-md">Code</dt-text>
  <dt-text>Default (inherits)</dt-text>
</dt-stack>
```

### Size

> Use `size` with `variant` to apply a raw font-size token while keeping the rest of the composition. `size` should not be used by itself; pair it with `variant`, or with legacy `kind` while migrating older usages.

Reviewer: ```suggestion
Using `size` with `variant` acts as an override of the `font-size` over its composed default. It applies a raw font-size token while keeping the rest of the composition. `size` will not work by itself, as it requires a pairing of `variant`, or with legacy `kind` when migrating older usages.
```

<dt-stack class="d-w100p d-ba d-bar-300 d-of-auto">
  <table class="d-w100p d-table">
    <tr class="d-va-baseline">
      <th>Size</th>
      <th>Output</th>
      <th>Example</th>
    </tr>
    <tr v-for="item in textSizeScale" :key="item.size" class="d-va-baseline">
      <th scope="row">
        <dt-text as="code" variant="code-xs" class="d-docsite-code">{{ item.size }}</dt-text>
      </th>
      <td>
        <dt-text variant="body-sm">{{ item.output }}</dt-text>
      </td>
      <td>
        <dt-text variant="body-md" :size="item.size">Text</dt-text>
      </td>
    </tr>
  </table>
</dt-stack>

```vue code-only
<dt-text
  variant="body-lg"
  size="300"

> >Body composition with a 20px font-size override</dt-text>

Reviewer: ```suggestion
<dt-text
  variant="body-xs"
  size="125"
>Body composition with font-size override  to fall between "body-xs" and "body-sm" </dt-text>
```
`body-lg`+`300` doesn't make sense since that's equivalent to just `body-md`. 

This example demos a relevant use case of "start with `body-xs`, but override with `size="125"` to give me a size that falls between `body-xs` and `body-sm`.

```

### Legacy kind

> The `kind` prop remains supported for backward compatibility. When `kind` is present and `variant` is absent, `size` keeps its legacy composition behavior.

Reviewer: ```suggestion
> [!WARNING] Favor using `variant` over `kind/size` combinations. The `kind` prop remains supported for backward compatibility. 

When `kind` is present and `variant` is absent, `size` keeps its legacy composition behavior.

`kind` is ignored if `variant` is present. 
```

<script setup>
import { computed, reactive } from 'vue';
import ExampleProfileCard from '@exampleComponents/ExampleProfileCard.vue';

const textSizeScale = [
  { size: 50, output: '10px' },
  { size: 75, output: '11px' },
  { size: 100, output: '12px' },
  { size: 125, output: '13px' },
  { size: 150, output: '14px' },
  { size: 200, output: '16px' },
  { size: 250, output: '18px' },
  { size: 300, output: '20px' },
  { size: 350, output: '22px' },
  { size: 400, output: '26px' },
  { size: 450, output: '29px' },
  { size: 500, output: '32px' },
  { size: 550, output: '36px' },
  { size: 600, output: '42px' },
  { size: 650, output: '46px' },
  { size: 700, output: '52px' },
  { size: 750, output: '58px' },

>   { size: 800, output: '66px' },

Reviewer: Let's use the existing `@data/type.json` source here instead of hard-coding the same scale

`fontSize.product` already has the stop + px mapping this table needs, so deriving `textSizeScale` from that would reduce (not solve) drift risk and keep this page aligned with the typography utility docs. 

It still doesn’t fully solve the underlying problem, since `type.json` itself is manually authored static data we should eventually automate from the token source too, among others. That's a different task; for this PR, using `@data/type.json` is acceptablestep of not adding another duplicate list.
